packages/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java

```diff
@@ -0,0 +1,212 @@
+        new MethodChannel(registrar.messenger(), "plugins.flutter.io/firebase_remote_config");
+    channel.setMethodCallHandler(new FirebaseRemoteConfigPlugin(channel));
+    sharedPreferences =
+        registrar.context().getSharedPreferences("FirebaseRCPlugin", Context.MODE_PRIVATE);
+  }
+
+  private FirebaseRemoteConfigPlugin(MethodChannel channel) {
+    this.channel = channel;
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, final Result result) {
+    switch (call.method) {
+      case "RemoteConfig#instance":
+        {
+          FirebaseRemoteConfigInfo firebaseRemoteConfigInfo =
+              FirebaseRemoteConfig.getInstance().getInfo();
+
+          Map<String, Object> properties = new HashMap<>();
+          properties.put("LAST_FETCH_TIME", firebaseRemoteConfigInfo.getFetchTimeMillis());
+          properties.put(
+              "LAST_FETCH_STATUS",
+              mapLastFetchStatus(firebaseRemoteConfigInfo.getLastFetchStatus()));
+          properties.put(
+              "IN_DEBUG_MODE",
+              firebaseRemoteConfigInfo.getConfigSettings().isDeveloperModeEnabled());
+          result.success(properties);
+          break;
+        }
+      case "RemoteConfig#setConfigSettings":
+        {
+          boolean debugMode = call.argument("debugMode");
+          final FirebaseRemoteConfig firebaseRemoteConfig = FirebaseRemoteConfig.getInstance();
+          FirebaseRemoteConfigSettings settings =
+              new FirebaseRemoteConfigSettings.Builder().setDeveloperModeEnabled(debugMode).build();
+          firebaseRemoteConfig.setConfigSettings(settings);
+          result.success(null);
+          break;
+        }
+      case "RemoteConfig#fetch":
+        {
+          long expiration =
+              call.argument("expiration") instanceof Integer
+                  ? Long.valueOf((Integer) call.argument("expiration"))
+                  : (Long) call.argument("expiration");
+          final FirebaseRemoteConfig firebaseRemoteConfig = FirebaseRemoteConfig.getInstance();
+          firebaseRemoteConfig
+              .fetch(expiration)
+              .addOnCompleteListener(
+                  new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                      FirebaseRemoteConfigInfo firebaseRemoteConfigInfo =
+                          firebaseRemoteConfig.getInfo();
+                      Map<String, Object> properties = new HashMap<>();
+                      properties.put(
+                          "LAST_FETCH_TIME", firebaseRemoteConfigInfo.getFetchTimeMillis());
+                      properties.put(
+                          "LAST_FETCH_STATUS",
+                          mapLastFetchStatus(firebaseRemoteConfigInfo.getLastFetchStatus()));
+                      if (!task.isSuccessful()) {
+                        final Exception exception = task.getException();
+                        channel.invokeMethod(
```

Comment:
I'm not sure why we are using `invokeMethod` to return to Dart. It seems like we could return these properties on the method invocation that called `"RemoteConfig#fetch"`. This would make awaiting a `fetch()` in Dart actually wait for the fetch to happen.

Reply:
This was used to updated some RemoteConfig properties on the Dart side, however this can be done in handling of the exception so removed this call to invokeMethod.